### PR TITLE
Support environment variables for all supported system properties in a uniform way

### PR DESCRIPTION
**Pull request by bertfrees**
See issue https://github.com/daisy/pipeline-assembly/issues/141

### CODE REVIEW

File: bom/pom.xml [hunk 1]
```diff
@@ -20,22 +20,22 @@
...
-    <daisy.persistence-mysql.version>2.0.0</daisy.persistence-mysql.version>
+    <daisy.persistence-derby.version>2.0.2-SNAPSHOT</daisy.persistence-derby.version>
+    <daisy.persistence-mysql.version>2.0.1-SNAPSHOT</daisy.persistence-mysql.version>
     <daisy.push-notifier.version>2.0.1</daisy.push-notifier.version>
-    <daisy.updater-wrapper.version>1.0.1</daisy.updater-wrapper.version>
-    <daisy.webservice.version>2.1.2</daisy.webservice.version>
-    <daisy.webservice-utils.version>2.2.2</daisy.webservice-utils.version>
+    <daisy.updater-wrapper.version>1.0.2-SNAPSHOT</daisy.updater-wrapper.version>
+    <daisy.webservice.version>2.1.3-SNAPSHOT</daisy.webservice.version>
+    <daisy.webservice-utils.version>3.0.0-SNAPSHOT</daisy.webservice-utils.version>
```
Comment (rdeltour) [inline bom/pom.xml]: I do not see API change in the `webservice` module, am I correct that the major change is due to change in the web service payload?

File: framework-core/src/main/java/org/daisy/pipeline/properties/Properties.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package org.daisy.pipeline.properties;
+
+import java.util.Map;
+
+public class Properties {
```
Comment (rdeltour) [inline framework-core/src/main/java/org/daisy/pipeline/properties/Properties.java]: if this class isn't intended to be extended, better make it `final` already, and if it's only intended as a static helper, we can add a private constructor too.

File: framework-core/src/main/java/org/daisy/pipeline/properties/Properties.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+	private final static java.util.Properties systemProperties = System.getProperties();
+	private final static Map<String,String> systemEnv = System.getenv();
```
Comment (rdeltour) [inline framework-core/src/main/java/org/daisy/pipeline/properties/Properties.java]: This works only if properties can't change dynamically. I think it's OK (?) but this should probably be documented in Javadoc?

File: framework-core/src/main/java/org/daisy/pipeline/properties/Properties.java (hunk 2 above)
Comment (bertfrees) [reply framework-core/src/main/java/org/daisy/pipeline/properties/Properties.java]: OK good point. I normally prefer`final abstract`, is that OK too?

File: framework-core/src/main/java/org/daisy/pipeline/properties/Properties.java (hunk 3 above)
Comment (bertfrees) [reply framework-core/src/main/java/org/daisy/pipeline/properties/Properties.java]: I didn't really care that much about dynamic changes when I wrote this. But if we can support it we probably should. So you say that the Properties object that `System.getProperties()` returns does not return a different value for `getProperty()` after a property was changed? In that case I should change it. And same question for `System.getenv()`?

File: bom/pom.xml (hunk 1 above)
Comment (bertfrees) [reply bom/pom.xml]: I don't remember TBH. Might be a mistake.

Comment (bertfrees) [reply bom/pom.xml]: I remember now. It's because I changed the `Properties` class into an enum.

Comment (rdeltour) [reply bom/pom.xml]: ok!

File: framework-core/src/main/java/org/daisy/pipeline/properties/Properties.java (hunk 2 above)
Comment (rdeltour) [reply framework-core/src/main/java/org/daisy/pipeline/properties/Properties.java]: OK.

File: framework-core/src/main/java/org/daisy/pipeline/properties/Properties.java (hunk 3 above)
Comment (rdeltour) [reply framework-core/src/main/java/org/daisy/pipeline/properties/Properties.java]: TBH I don't remember if `System.getProperties()` or `System.getenv()` are dynamic, so it needs to be checked first. If they aren't, then problem solved :)

Comment (bertfrees) [reply framework-core/src/main/java/org/daisy/pipeline/properties/Properties.java]: I checked, System.getProperties() is indeed dynamic. I'll add a comment. @rdeltour You meant that if they *are*, then problem solved, right?

File: framework-core/src/main/java/org/daisy/pipeline/properties/Properties.java (hunk 2 above)
Comment (bertfrees) [reply framework-core/src/main/java/org/daisy/pipeline/properties/Properties.java]: `final abstract` is not allowed. Normally I just add `abstract` without `final`. So your suggestion (`final` + private contructor) is the only real solution.

**Pull request closed without merging**